packages/samples/headless-react/src/components/search-status/search-status.class.tsx

```diff
@@ -0,0 +1,58 @@
+import {Component} from 'react';
+import {
+  buildSearchStatus,
+  SearchStatus as HeadlessSearchStatus,
+  SearchStatusState,
+  Unsubscribe,
+} from '@coveo/headless';
+import {engine} from '../../engine';
+
+export class SearchStatus extends Component {
+  private controller: HeadlessSearchStatus;
+  public state: SearchStatusState;
+  private unsubscribe: Unsubscribe = () => {};
+
+  constructor(props: {}) {
+    super(props);
+
+    this.controller = buildSearchStatus(engine);
+    this.state = this.controller.state;
+  }
+
+  componentDidMount() {
+    this.unsubscribe = this.controller.subscribe(() => this.updateState());
+  }
+
+  componentWillUnmount() {
+    this.unsubscribe();
+  }
+
+  private updateState() {
+    this.setState(this.controller.state);
+  }
+
+  render() {
```

Comment:
The render method could definitely be simplified by reducing indentation and doing only truthy checks, but doing so resulted in an order that felt difficult to read. This order seems more logical to me.

It would be possible to just print out the exact values of `hasResults`, `firstSearchExecuted` and `isLoading`, but I felt it important to represent every possible state (without any impossible state) since that's what I tend to look for when looking through examples.

Reply:
Maybe a small rewrite would be to represent those state with a function name ?

```ts
private get firstSearchIsExecuting() {
    const {hasResults, firstSearchExecuted, isLoading} = this.state;
    return !hasResults && !firstSearchExecuted && isLoading;
}

render() {
   if (this.firstSearchIsExecuting) {
       return <p>The first search ever is currently loading.</p>;
   }
}
```

Reply:
I think it more readable this way 👍

Reply:
Hmm, I feel like this introduces a less descriptive name (function name vs displayed text) and makes it so a developer reading this documentation would need to scroll up and down if they want more descriptive definitions.

If the objective is to see which values are `true` and which values are `false` in one line, then would the following kind of change be better?
```ts
render() {
    if (!hasResults && firstSearchExecuted && isLoading) {
        return <p>The first search ever is currently loading.</p>;
    }
    if (!hasResults && firstSearchExecuted && !isLoading) {
        return <p>No search was ever executed.</p>;
    }
    ...
    throw 'impossible state';
}
```

Reply:
This is better too, the original was the least readable with multiple indented ifs.
In Oli's example there is both the method and the text describing the state, and since the methods are small there's no scrolling up and down issue IMO.

I have no preference over both proposed solution, except the unhandled throw in the render method